draco/analysis/flagging.py

```diff
@@ -417,14 +417,71 @@ def process(self, data, mask):
             The input container with the weight dataset set to zero
             for samples that were identified as outliers.
         """
+        # Redistribute data over frequency
+        data.redistribute("freq")
+        mask.redistribute("freq")
 
+        # Multiply the weights by the inverse of the mask
         flag = ~mask.mask[:].view(np.ndarray)
 
         data.weight[:] *= flag.astype(np.float32)
 
         return data
 
 
+class MaskBeamformedWeights(task.SingleTask):
+    """Mask beamformed visibilities with anomalously large weights before stacking."""
```

Comment:
It's kind of obvious what it is, but ideally there would be a short docstring about the `nmed` attribute.